ocaml/tests/test_tar_ext.ml

```diff
@@ -0,0 +1,134 @@
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Helpers
+open Tar_ext
+
+let default_max_size = !Xapi_globs.bundle_file_max_size
+
+let ( // ) = Filename.concat
+
+let gen_test_file_script = "test_data/gen_non_cdn_test_file.sh"
+
+let test_file_dir = "/tmp/test_tar_ext"
+
+let unpack_dir = test_file_dir // "output"
+
+type test_case = {
+    description: string
+  ; test_file: string
+  ; max_size_limit: int64
+  ; expected: exn option
+}
+
+let test_cases =
+  [
+    {
+      description= "Test regular tar file"
+    ; test_file= "test_tar_ext_regular.tar"
+    ; max_size_limit= default_max_size
+    ; expected= None
+    }
+  ; {
+      description= "Test tar file with illegal path"
+    ; test_file= "test_tar_ext_illegal_path.tar"
+    ; max_size_limit= default_max_size
+    ; expected= Some (Illegal_file_path "test_illegal_dir/../file1.txt")
+    }
+  ; {
+      description= "Test tar file with unsupported file type"
+    ; test_file= "test_tar_ext_unsupported_file_type.tar"
+    ; max_size_limit= default_max_size
+    ; expected= Some (Unsupported_file_type "Symbolic")
+    }
+  ; {
+      description= "Test file exceeds max size limit"
+    ; test_file= "test_tar_ext_exceeds_max_size.tar"
+    ; max_size_limit= 500000L
+    ; expected=
+        Some
+          (File_exceeds_max_size_limit
+             "Current total unpacked file size: 1052672, exceeds max size \
+              limit: 500000"
+          )
+    }
+  ; {
+      description= "Test incomplete tar file"
+    ; test_file= "test_tar_ext_incomplete_file.tar"
+    ; max_size_limit= default_max_size
+    ; expected= Some (File_incomplete "Copied size is insistent with file size")
+    }
+  ; {
+      description= "Test end of file"
+    ; test_file= "test_tar_ext_end_of_file.tar"
+    ; max_size_limit= default_max_size
+    ; expected= Some (File_incomplete "End_of_file")
+    }
+  ; {
+      description= "Test corrupted tar file"
+    ; test_file= "test_tar_ext_corrupted_file.tar"
+    ; max_size_limit= default_max_size
+    ; expected= Some (File_corrupted "Tar.Header.Checksum_mismatch")
+    }
+  ; {
+      description= "Test file parsed failed"
+    ; test_file= "test_tar_ext_not_tar.tar"
+    ; max_size_limit= default_max_size
+    ; expected= Some (File_parsed_failed "Failure(\"Int64.of_string\")")
+    }
+  ]
+
+let test {test_file; max_size_limit; expected; _} () =
+  let run_command_with_output cmd =
+    let in_channel = Unix.open_process_in cmd in
+    try
+      while true do
+        let line = input_line in_channel in
+        print_endline line
+      done ;
+      close_in in_channel
+    with End_of_file -> close_in in_channel
+  in
+  run_command_with_output ("bash " ^ gen_test_file_script) ;
```

Comment:
For running shell code, using the `Bos` library would be simpler. See `pvs_proxy_setup.ml` where we are using it to run external commands and inspect the output and success.